Cargo.toml

```diff
@@ -81,6 +81,7 @@ url = "2.2"
 bytes = "1.0"
 serde = "1.0"
 serde_urlencoded = "0.7"
+proc-macro-hack = "0.5.19"
```

Comment:
What's this doing here?

Reply:
Merely because of [this](https://github.com/seanmonstar/reqwest/blob/master/.github/workflows/ci.yml#L233-L238).

In my branch there is a dependency with `proc-macro-hack=0.5.19`.

Using the minimum version (i.e., `proc-macro-hack = 0.5.0`), the latter crate fails to compile (i.e., `cargo check`).

It is still unclear to me why my change brings that dependency.
Maybe there is a not trivial dependency with some proc-macros.

Open to suggestions here :D

Reply:
> It is still unclear to me why my change brings that dependency

Ah I see, it's probably the other way around. Somebody who was already requiring an higher version of proc-macro-hack (probably https://github.com/rust-lang/futures-rs/pull/2520) removed the dependency on it, causing 0.5.0 to be eligible once again.

Reply:
I'm not sure of a good solution to this other than upgrading to cookie 0.16, which because of it using time 0.3 doesn't depend on proc-macro-hack at all.
If we have to stick with this workaround though I think it would be better to at least make proc-macro-hack an optional dependency and import it only with the cookie feature.

Reply:
Honestly, I don't see any better too.
Moved in the cookie optional dep.